brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/Alien4CloudEntitySpecFactory.java

```diff
@@ -67,18 +66,7 @@ public DefaultEntitySpecFactory(ManagementContext mgmt, ICSARRepositorySearchSer
         return spec;
     }
 
-    private boolean isComputeType(NodeTemplate nodeTemplate, Topology topology) {
-        return nodeTemplate.getType().equals(NormativeComputeConstants.COMPUTE_TYPE) ||
-                getIndexedNodeTemplate(nodeTemplate, topology)
-                        .getDerivedFrom()
-                        .contains(NormativeComputeConstants.COMPUTE_TYPE);
-    }
-
-    protected IndexedArtifactToscaElement getIndexedNodeTemplate(NodeTemplate nodeTemplate, Topology topology) {
-        return repositorySearchService.getRequiredElementInDependencies(
-                IndexedArtifactToscaElement.class,
-                nodeTemplate.getType(),
-                topology.getDependencies());
+    private boolean isComputeType(String nodeId, ToscaApplication toscaApplication) {
+        return alien4CloudFacade.isDerivedFrom(nodeId, toscaApplication, Alien4CloudFacade.COMPUTE_TYPE);
```

Comment:
I have quite a number of IDE warnings about unchecked types where you combine Facades and ToscaApplications. Can you tighten the generics? In this case because the class `implements EntitySpecFactory<Alien4CloudApplication>` you can replace `ToscaApplication` with `Alien4CloudApplication` and alter the constructor to `ToscaFacade<Alien4CloudApplication>`.